In [1]:
import pandas as pd


In [2]:
path = 'purchase_data.json'
purchase_data_df = pd.read_json(path)
purchase_data_df.head()

Age Gender  Item ID                                  Item Name  Price  \
0   38   Male      165                Bone Crushing Silver Skewer   3.37   
1   21   Male      119  Stormbringer, Dark Blade of Ending Misery   2.32   
2   34   Male      174                            Primitive Blade   2.46   
3   21   Male       92                               Final Critic   1.36   
4   23   Male       63                             Stormfury Mace   1.27   

             SN  
0     Aelalis34  
1        Eolo46  
2   Assastnya25  
3  Pheusrical25  
4        Aela59

In [3]:
num_players = len(purchase_data_df['SN'].unique())
num_players

573

In [4]:
unique_items = len(purchase_data_df['Item ID'].unique())
unique_items

183

In [5]:
avg_purchase_price = round(purchase_data_df['Price'].mean(), 2)
avg_purchase_price

2.93

In [6]:
num_purchases = purchase_data_df['Price'].size
num_purchases

780

In [7]:
tot_revenue = purchase_data_df['Price'].sum()
tot_revenue

2286.33

# Gender Demographics:


## What I need to know:

1. percentage and count of Male
2. percentage and count of Female
3. percentage and count of other / Non-Disclosed

## what I know:

1. there are more than one representation of some but not all players. the max representation is 5 times.
2. Collumn Gender contains the information needed however it has duplicates.
3. there are 573 unique players.
4. there are 780 rows in the dataframe.

## what I need to do:

1. drop the rows with duplicate players.
2. make new dataframe
3. groupby Gender
4. count gender groups
5. divide counts by total number of players to get percentages.

In [8]:
dropped_duplicate_SN = purchase_data_df.drop_duplicates(subset='SN',keep='first')
gender_grouped_df = dropped_duplicate_SN.groupby('Gender').count()
gender_renamed = gender_grouped_df.rename(index=str, columns={'Age': 'Gender Count', 'Item ID': 'Gender Percentage'})
gender_renamed['Gender Percentage'] = round((gender_renamed.loc[:,'Gender Percentage'] / num_players) * 100, 2)
gender_count = gender_renamed.loc[:, ['Gender Count', 'Gender Percentage']]
gender_count

Gender Count  Gender Percentage
Gender                                                
Female                          100              17.45
Male                            465              81.15
Other / Non-Disclosed             8               1.40

In [9]:
gender_groupby = purchase_data_df.groupby('Gender')

# Purchase Count by Gender:

In [10]:
count_df = gender_groupby.count()
count = count_df.loc[:,'Age']
purchase_count_df = count.to_frame()
purchase_count_df = purchase_count_df.rename(index=str, columns={'Age': 'Purchase Count'})
purchase_count_df
#purchase_count_df = count_df(index=str, columns={'Age': 'Purchase Count'})

Purchase Count
Gender                               
Female                            136
Male                              633
Other / Non-Disclosed              11

# Average Price by Gender:

In [11]:
avg_df = gender_groupby.mean()
avg = avg_df.loc[:,'Price']
avg_price_df = avg.to_frame()
avg_price_df = avg_price_df.rename(index=str, columns={'Price': 'Avg Purchase Price'})
avg_price_df

Avg Purchase Price
Gender                                   
Female                           2.815515
Male                             2.950521
Other / Non-Disclosed            3.249091

# Total Purchase Value by Gender:

In [18]:
total_df = gender_groupby.sum()
total = total_df.loc[:,'Price']
total_price_df = total.to_frame()
total_price_df = total_price_df.rename(index=str, columns={'Price': 'Total Purchase Value'})
total_price_df

Total Purchase Value
Gender                                     
Female                               382.91
Male                                1867.68
Other / Non-Disclosed                 35.74

# Normalized Totals by Gender:

In [21]:
normalized_price_df = total_price_df
normalized_price_df.iloc[0,0] = normalized_price_df.iloc[0,0] / purchase_count_df.iloc[0,0]
normalized_price_df.iloc[1,0] = normalized_price_df.iloc[1,0] / purchase_count_df.iloc[1,0]
normalized_price_df.iloc[2,0] = normalized_price_df.iloc[2,0] / purchase_count_df.iloc[2,0]
normalized_price_df.rename(index=str, columns={'Total Purchase Value': 'Normalized Purchase Value'}, inplace=True)
normalized_price

In [13]:


purchase_data_df.groupby('SN').sum().sort_values('Price', ascending=False)

Age  Item ID  Price
SN                                
Undirrala66    145      472  17.06
Saedue76       100      233  13.56
Mindimnya67    156      609  12.74
Haellysu29      63      353  12.73
Eoda93          66      284  11.58
Isursti83       72      299  11.05
Isurria36       54      390  11.01
Eusri70         90      325  10.55
Aerithllora36   87      281  10.45
Yasriphos60    120      237  10.40
Sondastan54    124      478  10.24
Haerith37       75      238  10.14
Chadjask77     108      363  10.01
Qarwen67        44      439   9.97
Sondim43        60      172   9.38
Tillyrin30      60      205   9.19
Lisistaya47     21      281   9.19
Tyisriphos58    76      282   9.18
Frichaya88      90      170   9.16
Assassasta79    34      226   9.12
Lisossan98      30      252   9.10
Yadanun74       75      373   9.09
Mindirra92      60      150   9.01
Aeliru63        44      203   8.98
Chanastsda67    45      299   8.96
Frichosiala98   75      312   8.94
Malunil62       38      201   8.89
Iskadarya95     30      200   8.79
Ilogha82        50      257   8.61
Yadaisuir65     44      115   8.56
...            ...      ...    ...
Rithe53         23       92   1.36
Pheusrical25    21       92   1.36
Isketo41        13        5   1.32
Tyaili86        17       86   1.28
Eollym91        23       86   1.28
Chamirra53      15       86   1.28
Ilaststa70      10       63   1.27
Aela59          23       63   1.27
Lassilsa63      23       63   1.27
Mindosiasya28   13      175   1.24
Raeduerin33     28      175   1.24
Heolo60         22      175   1.24
Ialidru40       19      175   1.24
Yaristi64       38      175   1.24
Ilrian97        13      175   1.24
Ryastycal90     20      122   1.21
Ilast79         20        6   1.20
Siasri67        22        6   1.20
Undirrasta74    22       41   1.16
Sundista85      22      123   1.14
Chamirrasya33   20       82   1.11
Eyircil84       23       74   1.06
Sondassa68      30       69   1.06
Lassadarsda57   24       25   1.03
Tridaira71      37       95   1.03
Rarith48        20       15   1.03
Alaesu91        18       95   1.03
Mindossa76      21       95   1.03
Lassista97      24       25   1.03
Ililsan66       18       15   1.03

[573 rows x 3 columns]